pkg/kubelet/kubelet.go

```diff
@@ -2134,6 +2134,13 @@ func (kl *Kubelet) HandlePodReconcile(pods []*v1.Pod) {
 		if status.NeedToReconcilePodReadiness(pod) {
 			mirrorPod, _ := kl.podManager.GetMirrorPodByPod(pod)
 			kl.dispatchWork(pod, kubetypes.SyncPodSync, mirrorPod, start)
+		} else if utilfeature.DefaultFeatureGate.Enabled(features.SidecarLifecycle) {
+			sidecarsStatus := status.GetSidecarsStatus(pod)
+			// If the non-sidecar containers are waiting and sidecars have become ready: trigger a sync
+			if sidecarsStatus.ContainersWaiting && sidecarsStatus.SidecarsPresent && sidecarsStatus.SidecarsReady {
```

Comment:
@tedyu this situation will occurs during initial pod startup, only once. the sidecars have started and now become ready, but the non sidecar containers will be in a waiting state, this check is used to trigger a sync so that we don't wait up to 1 minute between sidecars becoming ready and the containers starting